src/main/java/com/minecolonies/coremod/placementhandlers/RackPlacementHandler.java

```diff
@@ -50,14 +50,18 @@ public ActionProcessingResult handle(
             return ActionProcessingResult.SUCCESS;
         }
 
-        BlockEntity entity = world.getBlockEntity(pos);
-
         world.setBlock(pos, blockState, UPDATE_FLAG);
         if (tileEntityData != null)
         {
             handleTileEntityPlacement(tileEntityData, world, pos, settings);
         }
 
+        final BlockEntity entity  = world.getBlockEntity(pos);
+        if (entity instanceof TileEntityRack && !((TileEntityRack) entity).isSingle() && ((TileEntityRack) entity).getOtherChest() != null)
```

Comment:
It might be a good idea to update the block state even if the other chest is null.  The other rack in the pair may not be placed yet, or it may have been placed at different coordinates (rotation does not update the other-chest position).

In theory we should want all racks to revert to single when the first one is placed, and then update back to double when the second one is.